File: test-junit/src/test/java/io/micronaut/test/junit5/NonTransactionalTest.java
package io.micronaut.test.junit5;

import io.micronaut.context.ApplicationContext;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.transaction.test.DefaultTestTransactionExecutionListener;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

@MicronautTest(transactional = false)
@DbProperties
class NonTransactionalTest {

  @Inject
  ApplicationContext applicationContext;

  @Test
  void testSpringTransactionListenerMissing() {
    Assertions.assertFalse(applicationContext.containsBean(DefaultTestTransactionExecutionListener.class));
  }

}

File: test-kotest5/src/test/kotlin/io/micronaut/test/kotest/NonTransactionalTest.kt

package io.micronaut.test.kotest

import io.kotest.core.spec.style.BehaviorSpec
import io.kotest.matchers.shouldBe
import io.micronaut.context.ApplicationContext
import io.micronaut.test.extensions.kotest.annotation.MicronautTest
import io.micronaut.transaction.test.DefaultTestTransactionExecutionListener

@MicronautTest(transactional = false)
@DbProperties
class NonTransactionalTest(
        private val applicationContext: ApplicationContext) : BehaviorSpec({

    given("a test") {
        `when`("the test is not transactional") {
            then("the SpringTransactionTestExecutionListener does not exist") {
                applicationContext.containsBean(DefaultTestTransactionExecutionListener::class.java) shouldBe false
            }
        }
    }

})

